src/generator.rs

```diff
@@ -96,9 +97,50 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
         }
     }
 
+    let compressors: BTreeSet<&String> = devices
+        .iter()
+        .filter(|device| device.compression_algorithm.is_some())
+        .flat_map(|device| device.compression_algorithm.as_ref())
+        .collect::<BTreeSet<_>>();
+
+    if !compressors.is_empty() {
+        let known = get_known_compressors()?;
+
+        for comp in compressors.into_iter().filter(|&c| !known.contains(c)) {
+            let status = Command::new("modprobe")
+                .arg(format!("crypto-{}", comp))
+                .status()
+                .context("Failed to spawn modprobe")?;
+            if !status.success() {
+                warn!("modprobe crypto-{} failed, ignoring: code {}", comp, status);
+            }
+        }
+    }
+
     Ok(())
 }
 
+fn _get_known_compressors(path: &Path) -> Result<BTreeSet<String>> {
+    let content = fs::read_to_string(path).with_context(|| format!("Failed to read {:?}", path))?;
+
+    // Modules with type of `compression` have a `-generic` suffix in the driver name
```

Comment:
This comment isn't relevant anymore with your change from `driver` to `name` matching? Or keeping it for some context to anyone visiting this code and wondering what's going on? (I noticed my other comment for the now split function was rephrased to exclude mention that not all results would be compressor names)

```suggestion
    // Extracts the `name` field values from `/proc/crypto` output, some of which are compressor module names
```